mysql-test/suite/tianmu/t/issue682.test

```diff
@@ -0,0 +1,57 @@
+--source include/have_tianmu.inc
+
+use test;
+set sql_mode='STRICT_TRANS_TABLES,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+create table t1(a date not null);
```

Comment:
and as the mysql docuemnt said, disable in `NO_ZERO_DATE` mode, we should do some tests for this mode 
and `NO_ZERO_DATE` mode.